contracts/SDUtilityPool.sol

```diff
@@ -0,0 +1,771 @@
+            revert SDTransferFailed();
+        }
+        emit RequestRedeemed(msg.sender, sdToTransfer);
+    }
+
+    /**
+     * @notice Sender utilize SD from the pool to add it as collateral to run validators
+     * @param utilizeAmount The amount of the SD token to utilize
+     */
+    function utilize(uint256 utilizeAmount) external override whenNotPaused {
+        ISDCollateral sdCollateral = ISDCollateral(staderConfig.getSDCollateral());
+        (, , uint256 nonTerminalKeyCount) = sdCollateral.getOperatorInfo(msg.sender);
+        uint256 currentUtilizeSDCollateral = sdCollateral.operatorUtilizedSDBalance(msg.sender);
+        uint256 maxSDUtilizeValue = nonTerminalKeyCount * sdCollateral.convertETHToSD(maxETHWorthOfSDPerValidator);
+        if (currentUtilizeSDCollateral + utilizeAmount > maxSDUtilizeValue) {
+            revert SDUtilizeLimitReached();
+        }
+        accrueFee();
+        _utilize(msg.sender, utilizeAmount);
+    }
+
+    /**
+     * @notice utilize SD from the pool to add it as collateral for `operator` to run validators
+     * @dev only permissionless node registry contract can call
+     * @param operator address of an ETHx operator
+     * @param utilizeAmount The amount of the SD token to utilize
+     * @param nonTerminalKeyCount count of operator's non terminal keys
+     *
+     */
+    function utilizeWhileAddingKeys(
+        address operator,
+        uint256 utilizeAmount,
+        uint256 nonTerminalKeyCount
+    ) external override whenNotPaused {
+        UtilLib.onlyStaderContract(msg.sender, staderConfig, staderConfig.PERMISSIONLESS_NODE_REGISTRY());
+        ISDCollateral sdCollateral = ISDCollateral(staderConfig.getSDCollateral());
+        uint256 currentUtilizeSDCollateral = sdCollateral.operatorUtilizedSDBalance(operator);
+        uint256 maxSDUtilizeValue = nonTerminalKeyCount * sdCollateral.convertETHToSD(maxETHWorthOfSDPerValidator);
+        if (currentUtilizeSDCollateral + utilizeAmount > maxSDUtilizeValue) {
+            revert SDUtilizeLimitReached();
+        }
+        accrueFee();
+        _utilize(operator, utilizeAmount);
+    }
+
+    /**
+     * @notice Sender repays their own utilize
+     * @param repayAmount The amount to repay
+     */
+    function repay(uint256 repayAmount) external override whenNotPaused {
+        accrueFee();
+        _repay(msg.sender, repayAmount);
+    }
+
+    /**
+     * @notice Sender repays on behalf of utilizer
+     * @param repayAmount The amount to repay
+     */
+    function repayOnBehalf(address utilizer, uint256 repayAmount) external override whenNotPaused {
+        accrueFee();
+        _repay(utilizer, repayAmount);
+    }
+
+    /**
+     * @notice Sender repays on behalf of utilizer
+     * @dev only SD collateral contract can call
+     * @param repayAmount The amount to repay
+     */
+    function repayViaSDCollateral(address utilizer, uint256 repayAmount) external override whenNotPaused {
```

Comment:
Why doesn't this need an accrueFee() call?

Reply:
Earlier while fetching the balance of utilizer, we were calling accrueFee, hence was not required here but
removed this function, no use case now